src/super_gradients/training/metrics/pose_estimation_metrics.py

```diff
@@ -133,98 +134,128 @@ def reset(self) -> None:
+                            (For instance this is used in CrowdPose dataset)
+        :param gt_samples:  List of ground-truth samples
 
         """
-        predicted_poses, predicted_scores = self.post_prediction_callback(preds)  # Decode raw predictions into poses
+        predictions: List[PoseEstimationPredictions] = self.post_prediction_callback(preds)  # Decode raw predictions into poses
 
-        if gt_bboxes is None:
-            gt_bboxes = [compute_visible_bbox_xywh(torch.tensor(joints[:, :, 0:2]), torch.tensor(joints[:, :, 2])) for joints in gt_joints]
-
-        if gt_areas is None:
-            gt_areas = [bboxes[:, 2] * bboxes[:, 3] for bboxes in gt_bboxes]
+        if gt_samples is not None:
+            self._update_with_samples(predictions, gt_samples)
+        else:
+            self._update_with_old_style_args(predictions, gt_joints, gt_bboxes, gt_areas, gt_iscrowd)
 
-        if gt_iscrowd is None:
-            gt_iscrowd = [[False] * len(x) for x in gt_joints]
+    def _update_with_samples(self, predictions: List[PoseEstimationPredictions], gt_samples: List[PoseEstimationSample]):
+        for i in range(len(predictions)):
+            self.update_single_image(
+                predicted_poses=predictions[i].poses,
+                predicted_scores=predictions[i].scores,
+                gt_joints=gt_samples[i].joints,
+                gt_bboxes=gt_samples[i].bboxes,
+                gt_areas=gt_samples[i].areas,
+                gt_iscrowd=gt_samples[i].is_crowd,
+            )
 
-        for i in range(len(predicted_poses)):
+    def _update_with_old_style_args(
+        self,
+        predictions: List[PoseEstimationPredictions],
+        gt_joints: List[np.ndarray],
+        gt_bboxes: Optional[List[np.ndarray]],
+        gt_areas: Optional[List[np.ndarray]],
+        gt_iscrowd: Optional[List[np.ndarray]],
+    ):
+        for i in range(len(predictions)):
             self.update_single_image(
-                predicted_poses[i], predicted_scores[i], gt_joints[i], gt_areas=gt_areas[i], gt_bboxes=gt_bboxes[i], gt_iscrowd=gt_iscrowd[i]
+                predicted_poses=predictions[i].poses,
+                predicted_scores=predictions[i].scores,
+                gt_joints=gt_joints[i],
+                gt_bboxes=gt_bboxes[i] if gt_bboxes is not None else None,
+                gt_areas=gt_areas[i] if gt_areas is not None else None,
+                gt_iscrowd=gt_iscrowd[i] if gt_iscrowd is not None else None,
             )
 
     def update_single_image(
         self,
         predicted_poses: Union[Tensor, np.ndarray],
         predicted_scores: Union[Tensor, np.ndarray],
-        groundtruths: Union[Tensor, np.ndarray],
-        gt_bboxes: Union[Tensor, np.ndarray],
-        gt_areas: Union[Tensor, np.ndarray],
-        gt_iscrowd: Union[Tensor, np.ndarray, List[bool]],
+        gt_joints,
+        gt_bboxes,
+        gt_areas,
+        gt_iscrowd,
```

Comment:
Isnt the typing
```

        gt_joints: List[np.ndarray],
        gt_bboxes: List[np.ndarray],
        gt_areas: List[np.ndarray],
        gt_iscrowd: List[np.ndarray],
```
?

Reply:
Thanks. Fixed in d27f6245b5687c087a02e5143d760b9aa9ad2c06